### PR TITLE
Upgrade kafka version

### PR DESCRIPTION
**Pull request by eskabetxe**
Upgrade kafka version to last version

### CODE REVIEW

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/JsonEncoder.java [hunk 1]
```diff
@@ -1,44 +0,0 @@
...
-package io.prestosql.plugin.kafka.util;
-
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import kafka.serializer.Encoder;
-import kafka.utils.VerifiableProperties;
-
-import java.io.UncheckedIOException;
-
-public class JsonEncoder
```
Comment (eskabetxe) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/JsonEncoder.java:23]: this its not used and its same as JsonSerializer

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/NumberEncoder.java [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
- * limitations under the License.
- */
-package io.prestosql.plugin.kafka.util;
-
-import kafka.serializer.Encoder;
-import kafka.utils.VerifiableProperties;
-
-import java.nio.ByteBuffer;
-
-public class NumberEncoder
```
Comment (eskabetxe) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/NumberEncoder.java:21]: this its not used..
Encoder its previous to 2.0

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/JsonEncoder.java (hunk 1 above)
Comment (findepi) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/JsonEncoder.java:23]: please move this to a separate commit `Remove unused classes`

File: presto-kafka/pom.xml [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
+        <dep.kafka.version>2.4.1</dep.kafka.version>
```
Comment (findepi) [inline presto-kafka/pom.xml:17]: this is used once, so  no need for a property

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java [hunk 4]
```diff
@@ -34,7 +34,7 @@
 
     public TestingKafka()
     {
-        container = new KafkaContainer("5.2.1");
+        container = new KafkaContainer("5.4.1");
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java:37]: Is this related to kafka client update?

Comment (eskabetxe) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/TestingKafka.java:37]: its for version testing.. that its last version of confluent plataform, that have same kafka version of client

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/JsonEncoder.java (hunk 1 above)
Comment (eskabetxe) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/util/JsonEncoder.java:23]: done

File: presto-kafka/pom.xml (hunk 3 above)
Comment (eskabetxe) [reply presto-kafka/pom.xml]: done

**Comment by findepi** [comment]
LGTM

@eskabetxe do you have a particular reason for this upgrade?
or "just" fine desire to stay up to date?

**Comment by eskabetxe** [comment]
@findepi I prefer backward compatibility than forward compatibility :D
in this case this allow that client and server communicate with de highest Api version supported by both, taking advantage of new features introduced in the newest versions

[kafka protocol compatibility](https://kafka.apache.org/protocol#protocol_compatibility)

**Comment by findepi** [comment]
Merged, thanks!
